R/api-hc-hc_add_series.R

```diff
@@ -404,18 +412,23 @@ mutate_mapping <- function(data, mapping, drop = FALSE) {
   
   stopifnot(is.data.frame(data), inherits(mapping, "hcaes"), inherits(drop, "logical"))
   
-  # http://rmhogervorst.nl/cleancode/blog/2016/06/13/NSE_standard_evaluation_dplyr.html
+  # https://stackoverflow.com/questions/45359917/dplyr-0-7-equivalent-for-deprecated-mutate
+  # https://www.johnmackintosh.com/2018-02-19-theory-free-tidyeval/
+  
   tran <- as.character(mapping)
   newv <- names(mapping)
-
-  data <- dplyr::mutate_(data, .dots = setNames(tran, newv))
+  list_names <- setNames(tran, newv) %>% lapply(rlang::parse_quosure)
   
+  data <- dplyr::mutate(data, !!! list_names)
   # Reserverd  highcharts names (#241)
   if(has_name(data, "series"))
-    data <- rename_(data, "seriess" = "series")
+    #old <- "seriess"
+    #new <- "series"
+    data <- dplyr::rename(data, "seriess" = "series")
   
   if(drop)
-    data <- select_(data, .dots = names(mapping))
+    newv <- quos(names(mapping))
```

Comment:
@tungmilan 👍  good call  --

I realized while doing this that it would be better to use `rlang::syms()` vs `rlang::quos()`, as in the following in `muate_mapping`.  If using `rlang::quos()` and re-using `newv` I get the following error:  `Error: `newv` must resolve to integer column positions, not a list `. Anyway, I think `syms()` makes more sense anyway.
If you agree, I can push the version using `syms()` and we can test again :) 
```r
mutate_mapping <- function(data, mapping, drop = FALSE) {
  
  stopifnot(is.data.frame(data), inherits(mapping, "hcaes"), inherits(drop, "logical"))
  
  # https://stackoverflow.com/questions/45359917/dplyr-0-7-equivalent-for-deprecated-mutate
  # https://www.johnmackintosh.com/2018-02-19-theory-free-tidyeval/
  
  tran <- as.character(mapping)
  newv <- names(mapping)
  list_names <- setNames(tran, newv) %>% lapply(rlang::parse_quosure)
  
  data <- dplyr::mutate(data, !!! list_names)
  # Reserverd  highcharts names (#241)
  if(has_name(data, "series"))
    #old <- "seriess"
    #new <- "series"
    data <- dplyr::rename(data, "seriess" = "series")
  
  if(drop)
    newv <- rlang::syms(newv)
    data <- dplyr::select(data, !!! newv)
  
  data
  
}
```

Reply:
I like that. Let me know when it's ready for testing again :)